File: src/module/classify/13/513/2415.py
class Solution:
    def reverseOddLevels(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        q, level = [root], 0
        while q[0].left:
            q = list(chain.from_iterable((node.left, node.right) for node in q))
            if level == 0:
                for i in range(len(q) // 2):
                    x, y = q[i], q[len(q) - 1 - i]
                    x.val, y.val = y.val, x.val
            level ^= 1
        return root
    


class Solution:
    def reverseOddLevels(self, root: Optional[TreeNode]) -> Optional[TreeNode]:
        def dfs(node1: Optional[TreeNode], node2: Optional[TreeNode], is_odd_level: bool) -> None:
            if node1 is None: return
            if is_odd_level: node1.val, node2.val = node2.val, node1.val
            dfs(node1.left, node2.right, not is_odd_level)
            dfs(node1.right, node2.left, not is_odd_level)
        dfs(root.left, root.right, True)
        return root

File: src/module/classify/13/513/2415.js
var reverseOddLevels = function (root) {
  let queue = [];
  let depth = 0;
  if (root) {
    queue.push(root);
  }
  while (queue.length) {
    depth++;
    let size = queue.length;
    let nextQueue = [];
    for (let i = 0; i < size; i++) {
      let node = queue[i];
      if (node.left) nextQueue.push(node.left);
      if (node.right) nextQueue.push(node.right);
    }
    if (depth & 1) {
      console.log(queue[0].val);
      let n = nextQueue.length;
      size = n / 2;
      for (let i = 0; i < size; i++) {
        let j = n - i - 1;
        console.log(i, nextQueue[i].val);
        [nextQueue[i].val, nextQueue[j].val] ==
          [nextQueue[j].val, nextQueue[i].val];
      }
      if (nextQueue.length) {
        console.log(nextQueue[0].val);
      }
    }

    queue = nextQueue;
  }
  return root;
};
